gradio/component.py

```diff
@@ -95,3 +111,233 @@ def get_all_shortcut_implementations(cls):
+        """
+        return [], {}, True
+
+    def get_interpretation_scores(
+        self, x: Any, neighbors: List[Any], scores: List[float], **kwargs
+    ) -> List[Any]:
+        """
+        Arrange the output values from the neighbors into interpretation scores for the interface to render.
+        Parameters:
+        x (Any): Input to interface
+        neighbors (List[Any]): Neighboring values to input x used for interpretation.
+        scores (List[float]): Output value corresponding to each neighbor in neighbors
+        kwargs (Dict[str, Any]): Any additional arguments passed from get_interpretation_neighbors.
+        Returns:
+        (List[Any]): Arrangement of interpretation scores for interfaces to render.
+        """
+        pass
+
+    def generate_sample(self) -> Any:
+        """
+        Returns a sample value of the input that would be accepted by the api. Used for api documentation.
+        """
+        pass
+
+    # Output Functionalities
+    def postprocess(self, y):
+        """
+        Any postprocessing needed to be performed on function output.
+        """
+        return y
+
+    def deserialize(self, x):
+        """
+        Convert from serialized output (e.g. base64 representation) from a call() to the interface to a human-readable version of the output (path of an image, etc.)
+        """
+        return x
+
+
+class Textbox(Component):
+    """
+    Component creates a textbox for user to enter input. Provides a string as an argument to the wrapped function.
+    Input type: str
+    Demos: hello_world, diff_texts
+
+    Component creates a textbox to render output text or number.
+    Output type: Union[str, float, int]
+    Demos: hello_world, sentence_builder
+    """
+
+    def __init__(
+        self,
+        lines: int = 1,
+        placeholder: Optional[str] = None,
+        default: str = "",
+        label: Optional[str] = None,
+        numeric: Optional[bool] = False,
+        type: Optional[str] = "str",
+        optional: bool = False,
+    ):
+        """
+        Parameters:
+        lines (int): number of line rows to provide in textarea.
+        placeholder (str): placeholder hint to provide behind textarea.
+        default (str): default text to provide in textarea.
+        label (str): component name in interface.
+        numeric (bool): DEPRECATED.
+        type (str): DEPRECATED.
+        optional (bool): DEPRECATED
+        """
+        self.lines = lines
+        self.placeholder = placeholder
+        self.default = default
+        self.type = "str"
+        if numeric:
+            warnings.warn(
+                "The 'numeric' type has been deprecated. Use the Number component instead.",
+                DeprecationWarning,
+            )
+        if type:
+            warnings.warn(
```

Comment:
Remove this here. Because we actually _do need_ to store the `type` parameter

Instead, this should be:

```python
        self.type = type
```